69.md

```diff
@@ -0,0 +1,49 @@
+NIP-69
+======
+
+Source Control Context
+----------------------
+
+`draft` `optional` `author:garyKrause_`
+
+This NIP defines ways source control (git, etc.) context can be stored using nostr. Git is already decentralized, but the contexts around a git repository are largely centralized (PR comments, issue tracking, releases, etc). With the correct approach a nostr relay could store this context using many of the existing NIPs. This NIP defines a new kind, `kind: 69`, to represent a source control repository. This NIP purposely does not indicate git as a requirement but many of the concepts will be based around git methods of source control. The author hopes that the NIP is flexible enough to accommodate other source control methodologies.
+
+## Definition of a Source Control Event
+
+`kind: 69` uses tag the `b` tag to identify the base repository. The value of this tag SHALL be a list of remote source control repository URLs. These URLs MAY be a nostr relay URL or a traditional repository URL. The first URL in this list SHALL be considered the "default" repository URL for the context.
+
+The `content` of a Source Control Event MUST contain a text description or summary of the base repository. This enables clients that don't specifically handle this kind to display meaningful text in their UI.
+    
+```json
+{
+    ...
+  "kind": 69,
+  "tags": [
+    ["b", <repository url>, <repository url>, ...],
+  ],
+  "content": <summary or description>,
+  ...
+```
+
+A relay MAY host the repository along side it's nostr database which can be discovered using a standard `REQ` message filtering for `kind: 69`.
+
```

Comment:
I feel like I am missing something here. Would the repository be a standard ssh/http git repo? If so then why is this needed? The relay owner that want to run a git repo can just send a kind 69 note to the relay and it will show up.

```suggestion
```

Reply:
Good point. This was my way to describe a variation of the model but isn't needed in the NIP. Thank you.